operator/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Build the operator binary
-FROM --platform=${BUILDPLATFORM} golang:1.23 AS op-builder
+FROM golang:1.23 AS op-builder
```

Comment:
You want to keep this -- it lets you build multi-platform images natively. If you're having platform issues it might be because we're missing this on the other build targets.

Reply:
For whatever reason the `make docker-buildx` target was in the habit of adding `--platform=${BUILDPLATFORM}` automatically, but I see no reason to do that and changed it.